File: stepik-java-api/src/main/java/org/stepik/api/queries/StepikAbstractPostQuery.java
package org.stepik.api.queries;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;

/**
 * @author meanmail
 */
public abstract class StepikAbstractPostQuery<T> extends StepikAbstractQuery<T> {
    protected StepikAbstractPostQuery(@NotNull StepikAbstractAction stepikAction, @NotNull Class<T> responseClass) {
        super(stepikAction, responseClass, QueryMethod.POST);
    }

    @NotNull
    protected String getContentType() {
        return "application/json";
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/StepikAbstractPostQuery.kt
package org.stepik.api.queries

import org.stepik.api.actions.StepikAbstractAction

abstract class StepikAbstractPostQuery<T> protected constructor(stepikAction: StepikAbstractAction,
                                                                responseClass: Class<T>) :
        StepikAbstractQuery<T>(stepikAction, responseClass, QueryMethod.POST) {
    
    override val contentType: String = "application/json"
}
